x-pack/plugins/apm/public/components/app/service_overview/index.tsx

```diff
@@ -76,6 +82,10 @@ export function ServiceOverview() {
     ? 'column'
     : 'row';
 
+  if (!transactionType) {
+    return <NoTransactionsPrompt />;
+  }
```

Comment:
Thanks for adding this!  One nit: I think this will cause some layout shifts because `transactionType` will be `undefined` by default. We should only show the `NoTransactionsPrompt` when `transactionType` is `undefined` _and it was successfully loaded_ .

```suggestion
  if (!transactionType && transactionTypeStatus === FETCH_STATUS.SUCCESS) {
    return <NoTransactionsPrompt />;
  }
```

(`transactionTypeStatus` needs to be returned from `useApmServiceContext` in this case)